staging/src/k8s.io/kubectl/pkg/cmd/wait/wait.go

```diff
@@ -585,10 +585,14 @@ func (j JSONPathWait) isJSONPathConditionMet(event watch.Event) (bool, error) {
 // and check if it matches the desired condition
 func (j JSONPathWait) checkCondition(obj *unstructured.Unstructured) (bool, error) {
 	queryObj := obj.UnstructuredContent()
+	j.jsonPathParser.AllowMissingKeys(true)
```

Comment:
I would call `AllowMissingKeys(true)` when we create the parser [1] [2] since it changes the internal state of the parser and might produce unexpected side effects.

I wouldn't expect that `x` and `y` might be different in this case:
```golang
x, _ := j.jsonPathParser.FindResults(obj.UnstructuredContent())
j.checkCondition(obj)
y, _ := j.jsonPathParser.FindResults(obj.UnstructuredContent())
```
---
[1]
```golang
func newJSONPathParser(jsonPathExpression string) (*jsonpath.JSONPath, error) {
	j := jsonpath.New("wait").AllowMissingKeys(true)
	if jsonPathExpression == "" {
		return nil, errors.New("jsonpath expression cannot be empty")
	}
	if err := j.Parse(jsonPathExpression); err != nil {
		return nil, err
	}
	return j, nil
}
```

[2] https://github.com/kubernetes/kubernetes/blob/9c0eaddf121d91863b70b03cb408a7249b882e5e/staging/src/k8s.io/kubectl/pkg/cmd/wait/wait.go#L229-L238

Reply:
Great point!